website/project/views/node.py

```diff
@@ -897,13 +897,8 @@ def _get_user_activity(node, auth, rescale_ratio):
     # Counters
     total_count = len(node.logs)
 
-    # Note: It's typically much faster to find logs of a given node
-    # attached to a given user using node.logs.find(...) than by
-    # loading the logs into Python and checking each one. However,
-    # using deep caching might be even faster down the road.
-
     if auth.user:
-        ua_count = node.logs.find(Q('user', 'eq', auth.user)).count()
+        ua_count = NodeLog.find(Q('node', 'eq', node._id) & Q('user', 'eq', auth.user)).count()
```

Comment:
on `NodeLog`.
When doing a query on two keys, node and user in this case, mongo won't use individual indexes.
A compound index is required for optimizing this call.

```python
 __indices__ = [{
    'key_or_list': [
        ('user', 1),
        ('node', 1),
    ]
}]		
```
Should do the trick, but double check with robomongo or the mongo shell just to make sure.